vms/platformvm/block/executor/verifier.go

```diff
@@ -565,5 +569,70 @@ func (v *verifier) processStandardTxs(txs []*txs.Tx, feeCalculator fee.Calculato
 		}
 	}
 
+	// After processing all the transactions, deactivate any SoVs that might not
+	// have sufficient fee to pay for the next second.
+	//
+	// This ensures that SoVs are not undercharged for the next second.
+	err := deactivateLowBalanceSoVs(
+		v.txExecutorBackend.Config.ValidatorFeeConfig,
+		diff,
+	)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to deactivate low balance SoVs: %w", err)
+	}
+
 	return inputs, atomicRequests, onAcceptFunc, nil
 }
+
+// deactivateLowBalanceSoVs deactivates any SoVs that might not have sufficient
+// fees to pay for the next second.
+func deactivateLowBalanceSoVs(
+	config validatorfee.Config,
+	diff state.Diff,
+) error {
+	var (
+		accruedFees       = diff.GetAccruedFees()
+		validatorFeeState = validatorfee.State{
+			Current: gas.Gas(diff.NumActiveSubnetOnlyValidators()),
+			Excess:  diff.GetSoVExcess(),
+		}
+		potentialCost = validatorFeeState.CostOf(
+			config,
+			1, // 1 second
+		)
+	)
+	potentialAccruedFees, err := math.Add(accruedFees, potentialCost)
+	if err != nil {
+		return fmt.Errorf("could not calculate potentially accrued fees: %w", err)
+	}
+
+	// Invariant: Proposal transactions do not impact SoV state.
+	sovIterator, err := diff.GetActiveSubnetOnlyValidatorsIterator()
+	if err != nil {
+		return fmt.Errorf("could not iterate over active SoVs: %w", err)
+	}
+
+	var sovsToDeactivate []state.SubnetOnlyValidator
+	for sovIterator.Next() {
+		sov := sovIterator.Value()
+		// If the validator has exactly the right amount of fee for the next
+		// second we should not remove them here.
+		if sov.EndAccumulatedFee >= potentialAccruedFees {
+			break
```

Comment:
Should this be `continue`?

Reply:
No, I think `break` is correct.

This is iterating over `sov`s (which means that `sov.EndAccumulatedFee` is monotonically increasing). Once we find the first SoV that has `sov.EndAccumulatedFee >= potentialAccruedFees`, we know that all following SoVs will have `sov.EndAccumulatedFee >= potentialAccruedFees`.

Technically a `continue` would still work here - but it would mean that we are iterating over SoVs that do not need to be iterated over.

Reply:
Ah, I was missing that the sov's are ordered in that way. I think a comment explaining that any following sov's will also pass this check would be helpful.

Reply:
Added